cmds/core/io/cmos.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2012-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build amd64 386
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/u-root/u-root/pkg/cmos"
+	"github.com/u-root/u-root/pkg/memio"
+)
+
+func init() {
+	usageMsg += `io (cr index)... # read from CMOS register index [14-127]
+io (cw index value)... # write value to CMOS register index [14-127]
+io (rtcr index)... # read from RTC register index [0-13]
+io (rtcw index value)... # write value to RTC register index [0-13]
+`
+	readCmds["cr"] = cmd{cmosRead, 7, 8}
+	readCmds["rtcr"] = cmd{rtcRead, 7, 8}
+	writeCmds["cw"] = cmd{cmosWrite, 7, 8}
+	writeCmds["rtcw"] = cmd{rtcWrite, 7, 8}
+}
+
+func cmosRead(reg int64, data memio.UintN) error {
+	regVal := memio.Uint8(reg)
+	if regVal < 14 {
+		return fmt.Errorf("byte %d is inside the range 0-13 which is reserved for RTC", regVal)
```

Comment:
yay! an error message! Linux just gives you EINVAL on stuff like this, it's nice to have a meaningful error :-)